src/audiomixerboard.cpp

```diff
@@ -1305,6 +1308,137 @@ void CAudioMixerBoard::SetAllFaderLevelsToNewClientLevel()
     }
 }
 
+void CAudioMixerBoard::AutoAdjustAllFaderLevels()
+{
+    QMutexLocker locker ( &Mutex );
+
+    // initialize variables used for statistics
+    float vecMinLevel[MAX_NUM_FADER_GROUPS + 1];
+    float vecMaxLevel[MAX_NUM_FADER_GROUPS + 1];
+    int   vecChannelsPerGroup[MAX_NUM_FADER_GROUPS + 1];
+    for ( int i = 0; i < MAX_NUM_FADER_GROUPS + 1; ++i )
+    {
+        vecMinLevel[i] = UPPER_BOUND_SIG_METER;
+        vecMaxLevel[i] = LOW_BOUND_SIG_METER;
+        vecChannelsPerGroup[i] = 0;
+    }
+
+    // compute min/max level per group and number of channels per group
+    for ( int i = 0; i < MAX_NUM_CHANNELS; ++i )
+    {
+        // only apply to visible faders (and not to my own channel fader)
+        if ( vecpChanFader[i]->IsVisible() && ( i != iMyChannelID ) )
+        {
+            // map averaged meter output level to decibels
+            // (invert CStereoSignalLevelMeter::CalcLogResultForMeter)
+            float leveldB = vecAvgLevels[i] *
+                ( UPPER_BOUND_SIG_METER - LOW_BOUND_SIG_METER ) /
+                NUM_STEPS_LED_BAR + LOW_BOUND_SIG_METER;
+
+            int group = vecpChanFader[i]->GetGroupID();
+            if ( group == INVALID_INDEX )
+                group = 4;
+
+            if ( leveldB >= AUTO_FADER_NOISE_THRESHOLD_DB )
+            {
+                vecMinLevel[group] = fmin ( vecMinLevel[group], leveldB );
+                vecMaxLevel[group] = fmax ( vecMaxLevel[group], leveldB );
+            }
+            ++vecChannelsPerGroup[group];
+        }
+    }
+
+    // compute the number of active groups (at least one channel)
+    int cntActiveGroups = 0;
+    for ( int i = 0; i < MAX_NUM_FADER_GROUPS + 1; ++i )
+    {
+        cntActiveGroups += vecChannelsPerGroup[i] > 0;
+    }
+
+    // only my channel is active, nothing to do
+    if ( cntActiveGroups == 0 )
```

Comment:
Style - always wrap in braces:
```
if ( X ) {
  return;
}
```

Reply:
Ok. I'll place the first bracket on the next line (after `if` line) since this seems to be the style in the majority of cases.

By the way: Shall I _resolve conversation_ or will you do so? (also applies for the other reviews)

Reply:
Oops, yes - next line...  (yes... really... plenty of white space... it's cheap... :) )

Reply:
Have you pushed this change?  (For some reason Github hasn't flagged the file change since I viewed it.)

Reply:
Now I have committed it after using rebasing from the current upstream master.